src/DynamoCoreWpf/Utilities/NodeContextMenuBuilder.cs

```diff
@@ -0,0 +1,453 @@
+                    Path = new PropertyPath(nameof(NodeViewModel.SetLacingTypeCommand)),
+                },
+                commandParameter: "CrossProduct",
+                isCheckable: true,
+                isChecked: new Binding
+                {
+                    Source = NodeViewModel,
+                    Path = new PropertyPath(nameof(NodeViewModel.ArgumentLacing)),
+                    Converter = new EnumToBooleanConverter(),
+                    ConverterParameter = "CrossProduct",
+                    Mode = BindingMode.OneWay
+                }
+            ));
+
+            return lacingMenuItem;
+        }
+
+        // To be connected in a future PR for Node Info States.
+        private static MenuItem BuildDismissedAlertsMenuItem()
+        {
+            //MenuItem dismissedAlertsMenuItem = CreateMenuItem
+            //(
+            //    name: "dismissedAlerts",
+            //    header: Wpf.Properties.Resources.NodeInformationalStateDismissedAlerts,
+            //    itemsSource: new Binding
+            //    {
+            //        Source = viewModel,
+            //        Path = new PropertyPath(nameof("DismissedAlerts"),
+            //        UpdateSourceTrigger = UpdateSourceTrigger.PropertyChanged
+            //    }
+            //);
+            //DataTemplate itemTemplate = new DataTemplate(typeof(MenuItem));
+
+            //FrameworkElementFactory grid = new FrameworkElementFactory(typeof(Grid));
+            //grid.Name = "mainTemplateGrid";
+            //grid.SetValue(WidthProperty, 220.0);
+            //grid.SetValue(HeightProperty, 30.0);
+            //grid.SetValue(MarginProperty, new Thickness(-15,0,0,0));
+            //grid.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Left);
+            //grid.SetValue(VerticalAlignmentProperty, VerticalAlignment.Stretch);
+            //grid.SetValue(BackgroundProperty, new SolidColorBrush(Colors.Transparent));
+
+            //FrameworkElementFactory textBlock = new FrameworkElementFactory(typeof(TextBlock));
+            //textBlock.SetValue(HorizontalAlignmentProperty, HorizontalAlignment.Stretch);
+            //textBlock.SetValue(MarginProperty, new Thickness(15, 0, 0, 0));
+            //textBlock.SetValue(IsHitTestVisibleProperty, false);
+            //textBlock.SetValue(VerticalAlignmentProperty, VerticalAlignment.Center);
+            //textBlock.SetValue(ForegroundProperty, new SolidColorBrush(Color.FromRgb(238,238,238)));
+
+            //dismissedAlertsMenuItem.ItemTemplate = itemTemplate;
+
+            //AddContextMenuItem(dismissedAlertsMenuItem, insertionPoint++);
+
+            return null;
+        }
+
+        internal static void AddInjectedNodeViewCustomizationMenuItems(OrderedDictionary nodeViewCustomizationMenuItems)
+        {
+            foreach (DictionaryEntry keyValuePair in nodeViewCustomizationMenuItems)
+            {
+                ContextMenu.Items.Add(keyValuePair.Value);
+            }
+        }
```

Comment:
@mjkkirschner I think in the latest version of the files there's a DictionaryEntry in this foreach loop. Not entirely sure why the PR isn't showing this on the main page! Must be outdated.